index.html

```diff
@@ -1787,6 +1935,296 @@ <h2>
+          graphical representations of a [=file type=] on a platform. This MAY
+          be replaced by the user agent for privacy & security reasons.
+        </p>
+      </section>
+      <section>
+        <h2>
+          Processing file handler items
+        </h2>
+        <p>
+          To <dfn>process a file handler item</dfn>, given [=ordered map=]
+          |item:ordered map|, [=URL=] |start_url:URL|, [=URL=] |scope:URL|, and
+          [=URL=] |manifest URL:URL|.
+        </p>
+        <ol class="algorithm">
+          <li>Return failure if it's the case that:
+            <ul>
+              <li>|item| is not [=ordered map=].
+              </li>
+              <li>|item|["action"] doesn't [=map/exist=] or is not a
+              [=string=].
+              </li>
+              <li>|item|["name"] [=map/exists=] and is not a [=string=].
+              </li>
+              <li>|item|["accept"] doesn't [=map/exist=].
+              </li>
+              <li>|item|["accept"] is not a [=dictionary=].
+              </li>
+              <li>|item|["accept"] is [=list/empty].
+              </li>
+            </ul>
+          </li>
+          <li>Let |url:URL| be the result of [=URL Parser|parsing=]
+          |item|["action"] with |start_url URL| as the base URL.
+          </li>
+          <li>If |url| is failure, return failure.
+          </li>
+          <li>If |url| is not [=manifest/within scope=] of |scope|, return
+          failure.
+          </li>
+          <li>Let |accept:ordered map| be an [=ordered map=].
+          </li>
+          <li>[=map/iterate=] |mime_type_string:string| → |extensions| of
+          |item|["accept"]
+            <ol>
+              <li>If |extensions| is not a [=string=] or [=list=], return
+              failure.
+              </li>
+              <li>If |extensions| is the empty string or an [=list/empty=]
+              [=list=], return failure.
+              </li>
+              <li>If |extensions| is a [=list=] with items that are not
+              [=string=]s, return failure.
+              </li>
+              <li>Let |mime_type_parsed:mime type| be the result of running the
+              steps of [=parse a mime type=] on |mime_type_string|.
+              </li>
+              <li>If |mime_type_parsed|.type is not listed in as a top-level
+              type in [[IANA_MEDIA_TYPES]], return failure.
+              </li>
+              <li>Add |mime_type_string| → |extensions| to |accept|.
+              </li>
+            </ol>
+          </li>
+          <li>Let |file_handler:ordered map| be |ordered map| «[ "action" →
+          |url|, "name" → |item|["name"], "accept" → |accept| ]».
+          </li>
+          <li>[=Process image resources=] passing |item|["icons"],
+          |file_handler|, |manifest URL|, and "icons".
+          </li>
+          <li>The user agent MAY replace |file_handler|["icons"] with the
+          application icon.
```

Comment:
Should it be this specific?

One proposal that has floated around before has been to use the application icon badged with the file type icon.

Reply:
Agreed. I'll just say something vague.